vroom/pluginextensions.vroom

```diff
@@ -0,0 +1,200 @@
+  :call g:registry.AddExtension(g:foo_extension)
+
+If the original plugin has registered a validator (as is the case here), the
+validator will be called to verify the extension.  Any errors returned from
+the validator will be shouted to the user (and will prevent the extension
+from being registered), but will not be propagated to the caller.
+
+TODO: would it make sense for AddExtension() to return whether it succeed,
+then?
+
+TODO: should we define what exception the validator should throw (or even
+provide a richer interface for indicating validation failures), or is it
+reasonable (as here) to just delegate everything to the validator? (which
+means that e.g. we can't in general rely on any particular error being
+shouted)
+
+  :let g:registry = maktaba#extension#GetRegistry('myplugin')
+  :let g:bad_extension = {'command': 'bad-format'}
+  :call g:registry.AddExtension(g:bad_extension)
+  ~ ERROR(BadValue): No filetype.
+  :let g:bar_extension = {'filetype': 'bar', 'command': 'bar-format'}
+  :call g:registry.AddExtension(g:bar_extension)
+
+The original plugin can now retrieve the list of extensions from its
+extension registry.
+
+  :let g:registry = g:plugin.GetExtensionRegistry()
+  :echomsg string(g:registry.GetExtensions())
+  ~ [{'filetype': 'foo', 'command': 'foo-format'},
+  | {'filetype': 'bar', 'command': 'bar-format'}]
+
+The list of extensions is returned in the order that AddExtension() is
+called, with one exception, noted below.
+
+
+Note that the registry object returned by maktaba#extension#GetRegistry() is
+limited compared to the one returned by GetExtensionRegistry() in that the
+SetValidator() and GetExtensions() functions are not implemented.
+
+  :let g:registry = maktaba#extension#GetRegistry('myplugin')
+  :call maktaba#error#Try(maktaba#function#Method(g:registry, 'GetExtensions'))
+  ~ ERROR(NotImplemented): Not accessible by external plugins.
+  :call maktaba#error#Try(maktaba#function#Method(g:registry, 'SetValidator')
+  |.WithArgs('MyFormatterValidator'))
+  ~ ERROR(NotImplemented): Not accessible by external plugins.
+
+Extensions can also be registered by the plugin providing the extension
+point, which can used as a way of providing default implementations that
+other plugins can override.
+
+This is done simply by calling AddExtension(), as usual.
+
+  :let g:registry = g:plugin.GetExtensionRegistry()
+  :let g:baz_extension = {'filetype': 'baz', 'command': 'baz-format'}
+  :call g:registry.AddExtension(g:baz_extension)
+  :echomsg string(g:registry.GetExtensions())
+  ~ [{'filetype': 'baz', 'command': 'baz-format'},
+  | {'filetype': 'foo', 'command': 'foo-format'},
+  | {'filetype': 'bar', 'command': 'bar-format'}]
+
+There is one important difference here: extensions added via the internal
+interface will always precede those added via the external interface.
+
+TODO: should internally-added extensions have an _is_internal boolean, or is
```

Comment:
You mentioned _is_internal in one of the original comments on #130.  I've removed this TODO for now, but that's where it came from.